File: examples/server/tornado/latency.py
import os

import tornado.ioloop
import tornado.web
from tornado.options import define, options, parse_command_line

import socketio

define("port", default=5000, help="run on the given port", type=int)
define("debug", default=False, help="run in debug mode")

sio = socketio.AsyncServer(async_mode="tornado")


class MainHandler(tornado.web.RequestHandler):
    def get(self):
        self.render("latency.html")


@sio.event
async def ping_from_client(sid):
    await sio.emit("pong_from_server", room=sid)


def main():
    parse_command_line()
    app = tornado.web.Application(
        [
            (r"/", MainHandler),
            (r"/socket.io/", socketio.get_tornado_handler(sio)),
        ],
        template_path=os.path.join(os.path.dirname(__file__), "templates"),
        static_path=os.path.join(os.path.dirname(__file__), "static"),
        debug=options.debug,
    )
    app.listen(options.port)
    tornado.ioloop.IOLoop.current().start()


if __name__ == "__main__":
    main()

File: examples/server/javascript/latency.js
const express = require('express');
const { createServer } = require("http");
const { Server } = require("socket.io");

const app = express();
const httpServer = createServer(app);
const io = new Server(httpServer);

const port = process.env.PORT || 5000;

app.use(express.static(__dirname + '/latency_public'));

io.on('connection', socket => {
  console.log(`connect ${socket.id}`);

  socket.on('ping_from_client', () => {
    socket.emit('pong_from_server');
  });

  socket.on('disconnect', () => {
    console.log(`disconnect ${socket.id}`);
  });
});

httpServer.listen(port, () => console.log(`server listening on port ${port}`));
